realm/realm-library/src/main/java/io/realm/ManagedSetManager.java

```diff
@@ -158,14 +159,16 @@ public SetValueOperator(BaseRealm baseRealm, OsSet osSet) {
     }
 
     public boolean add(@Nullable E e) {
-        boolean alreadyExisted = osSet.contains(e);
+        boolean found = osSet.contains(e);
```

Comment:
I guess this is a bit expensive. Maybe raise with core to get the information as part of the `add` lower down the stack.

Reply:
Yes, this was me not being aware that the OS returns a Pair<Boolean, Index> which can be used for this very purpose 🤦 It is changed in one of the next PRs - sorry about it 🙊